### DISCUSSION TITLE
wait until block implementation

### DISCUSSION THREAD

**Discussion by Sunny-da-moth** [TurboWarp/extensions#582] [Q&A] [answered]
on the developper doc there is an example of async blocks, but i didnt get how to make a wait until block, chatgpt didnt help as every version of the script he made either didnt wait or only checked if the condition was true once. if you were to explain what functions to use or provide a working example i'd greatly apreaciate it

**Comment by LilyMakesThings** [comment]
You can check the scratch-vm (it's a TurboWarp repository) however the last time I tried to reimplement the block, it didn't work in the compiler.

**Comment by Sunny-da-moth** [comment]
```
waitUntil(args, util) {
  const condition = Scratch.Cast.toBoolean(args.CONDITION);
  if (!condition) {
    util.yield();
  }
}
```
this is the implementation of wait until block used in vm, when i reused it for the extension it appears to only check the condition once when its activated and no more so it either waits forever or doesnt wait at all

**Reply by LilyMakesThings** [reply, marked as answer]
As I said in my previous comment, this is a problem with the compiler. If you turn of the compiler it will work fine.
